test/usertest.spec.js

```diff
@@ -1,14 +1,168 @@
 import chai, { expect } from 'chai';
 import chaiHttp from 'chai-http';
+import { afterEach, beforeEach, describe } from 'mocha';
+import { Op } from 'sequelize';
 import app from '../src/app';
+import models from '../src/database/models';
 
 chai.should();
 
 chai.use(chaiHttp);
 
-describe('prime test', () => {
-  it('Should fetch data from the database', async () => {
-    const res = await chai.request(app).get('/api/v2/test').send();
-    expect(res).to.have.status(200);
+const user = {
+  UserName: 'kevi',
+  Email: 'didas@andela.com',
+  Password: 'Andelahfghfh',
+  FirstName: 'JohnS',
+  LastName: 'DoeS',
+};
+
+const user3 = {
+  UserName: 'kevinNew',
+  Email: 'you@test.com',
+  Password: 'Password@2022',
+  FirstName: 'JohnS',
+  LastName: 'DoeS',
+};
+
+const realUser = {
+  UserName: 'kevinNew',
+  Email: 'itiswasp@gmail.com',
+  Password: 'Password@2022',
+  FirstName: 'JohnS',
+  LastName: 'DoeS',
+};
+
+describe('POST REGISTER', () => {
+  describe('POST /api/auth/register', () => {
+    beforeEach((done) => {
+      models.User.destroy({
+        where: { email: { [Op.not]: [realUser.Email] } },
+      }).then(() => done());
+    });
+    it('It Give an error on Validation of Password', (done) => {
+      chai
+        .request(app)
+        .post('/api/auth/register')
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          res.body.should.have
+            .property('message')
+            .eql(
+              'Password must contain atleast a number, a special character, an upper-case letter and longer than 8 characters'
+            );
+          done();
+        })
+        .timeout(100000);
+    });
+
+    it('It Give an error on Validation of UserName', (done) => {
+      chai
+        .request(app)
+        .post('/api/auth/register')
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          res.body.should.have.property('message');
+          done();
+        })
+        .timeout(100000);
+    });
+
+    it('It Give an error on Validation of Email', (done) => {
+      chai
+        .request(app)
+        .post('/api/auth/register')
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          res.body.should.have.property('message');
+          done();
+        })
+        .timeout(100000);
+    });
+    it('It Give an error on Validation of FirstName', (done) => {
+      chai
+        .request(app)
+        .post('/api/auth/register')
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          res.body.should.have.property('message');
```

Comment:
Same as the previous validation errors, you should assert the value of the message returned